pkg/sidecar/sidecar.go

```diff
@@ -153,6 +155,22 @@ func (s *Sidecar) updateCertificates(svidResponse *workloadapi.X509Context) {
 // signalProcess sends the configured Renew signal to the process running the proxy
 // to reload itself so that the proxy uses the new SVID
 func (s *Sidecar) signalProcess() (err error) {
+	if s.config.PidFileName != "" {
```

Comment:
if `s.config.Cmd` is empty but `s.config.PidFileName` is non-empty it can fall through to the next statement. Previously it wouldn't because the if statement above would've stopped it. 

I think we can remove the if statement on 143 that gates this function call and then add the `if s.config.Cmd != ""` inside the case nil at line 176.